packages/hardhat-core/src/internal/core/config/config-resolution.ts

```diff
@@ -154,12 +155,18 @@ function resolveHardhatNetworkConfig(
 
   const mining = resolveMiningConfig(hardhatNetworkConfig.mining);
 
+  const minGasPrice = new BN(
+    hardhatNetworkConfig.minGasPrice ??
+      clonedDefaultHardhatNetworkParams.minGasPrice
+  );
+
   const config = {
     ...clonedDefaultHardhatNetworkParams,
     ...hardhatNetworkConfig,
     accounts,
     forking,
     mining,
+    minGasPrice,
```

Comment:
This is the kind of change produced by the divergence between the `HardhatNetworkUserConfig` types and the `HardhatNetworkConfig` types. If they had the same type, then passing `minGasPrice` like this wouldn't be necessary.

Reply:
yeah, but i think this is a good trade off